git-service/src/softserve/handlers.rs

```diff
@@ -1,32 +1,103 @@
+    if check_user_exists(&user.username)? {
+        warn!("Attempt to create existing user: {}", user.username);
+        return Err(GitServiceError::UserAlreadyExists(user.username.clone()));
     }
+
+    create_user(&user.username)
+        .map(|response| {
+            info!("User created: {}", user.username);
+            Ok(HttpResponse::Ok().json(response))
+        })
+        .map_err(|e| {
+            error!("Failed to create user {}: {}", user.username, e);
+            GitServiceError::FailedToCreateUser(e.to_string())
+        })?
 }
 
-pub async fn handle_create_token(token_req: web::Json<CreateTokenRequest>) -> impl Responder {
-    match create_token(&token_req.token_name) {
-        Ok(token) => HttpResponse::Ok().json(CreateTokenResponse {
-            token_name: token_req.token_name.clone(),
-            token: token.trim().to_string(),
-        }),
-        Err(e) => HttpResponse::InternalServerError().body(format!("Error: {}", e)),
-    }
+pub async fn handle_create_token(
+    token_req: web::Json<CreateTokenRequest>,
+) -> Result<HttpResponse, GitServiceError> {
+    create_token(&token_req.token_name)
+        .map(|response| {
+            info!("Token created: {}", token_req.token_name);
+            Ok(HttpResponse::Ok().json(response))
+        })
+        .map_err(|e| {
+            error!("Failed to create token {}: {}", token_req.token_name, e);
+            GitServiceError::FailedToCreateToken(e.to_string())
+        })?
 }
 
-pub async fn test_connection() -> impl Responder {
-    match create_user("test_user") {
-        Ok(_) => HttpResponse::Ok().body("Connection to Soft Serve successful"),
-        Err(e) => HttpResponse::InternalServerError().body(format!("Connection failed: {}", e)),
+pub fn check_repo_exists(repo_name: &str) -> Result<bool, GitServiceError> {
+    let command = "repo list";
+    execute_command(command)
+        .map(|output| {
+            let repos: Vec<&str> = output.lines().collect();
+            repos.iter().any(|&repo| repo.trim() == repo_name)
+        })
+        .map_err(|e| {
+            error!("Failed to check repo existence: {}", e);
+            GitServiceError::FailedToCheckRepoExistence(e.to_string())
+        })
+}
+
+pub fn check_user_exists(username: &str) -> Result<bool, GitServiceError> {
+    let command = "user list";
+    execute_command(command)
+        .map(|output| {
+            let users: Vec<&str> = output.lines().collect();
+            users.iter().any(|&user| user.trim() == username)
+        })
+        .map_err(|e| {
+            error!("Failed to check user existence: {}", e);
+            GitServiceError::FailedToCheckUserExistence(e.to_string())
+        })
+}
+
+pub async fn handle_create_repo(
+    repo_req: web::Json<CreateRepoRequest>,
+) -> Result<HttpResponse, GitServiceError> {
+    if check_repo_exists(&repo_req.repo_name)? {
+        warn!("Attempt to create existing repo: {}", repo_req.repo_name);
```

Comment:
This log should be an error and not a warning since it returns an error
```suggestion
        error!("Attempt to create an existing repo: {}", repo_req.repo_name);
```